detox/src/devices/allocation/drivers/android/emulator/launchEmulatorProcess.js

```diff
@@ -46,6 +31,8 @@ function launchEmulatorProcess(emulatorName, emulatorExec, emulatorLaunchCommand
 
   log = log.child({ child_pid: childProcessPromise.childProcess.pid });
 
+  adb.waitForDevice(adbName).then(() => childProcessPromise._cpResolve());
```

Comment:
What happens if something indeed goes wrong here? Emulator crash or whatever? Will it throw or hang forever?